channel/pools.go

```diff
@@ -419,6 +441,72 @@ func (pool *lifePool) mobDamaged(poolID int32, damager *Player, dmg ...int32) {
 	}
 }
 
+func (pool *lifePool) handleMobSummon(summoner *monster, skillLevel byte, skillData nx.MobSkill) {
+	if len(skillData.MobID) == 0 {
+		return
+	}
+
+	minY := summoner.pos.y - 100
+	maxY := summoner.pos.y + 100
+	minX := summoner.pos.x - 200
+	maxX := summoner.pos.x + 200
+
+	randX := minX
+	if maxX > minX {
+		randX = minX + int16(pool.rNumber.Intn(int(maxX-minX)))
+	}
+
+	for i, mobID := range skillData.MobID {
+		summonY := minY
+		if maxY > minY {
+			summonY = minY + int16(pool.rNumber.Intn(int(maxY-minY)))
+		}
+
+		offset := int16(0)
+		if i > 0 {
+			if i%2 == 1 {
+				offset = int16(35 * (i + 1) / 2)
+			} else {
+				offset = int16(-40 * (i / 2))
+			}
+		}
+		summonX := randX + offset
+		spawnPos := newPos(summonX, summonY, summoner.pos.foothold)
+
+		err := pool.spawnMobFromID(
```

Comment:
This function sends a mob show packet so setting the summon properties further down won't affect the spawn.

https://github.com/Hucaru/Valhalla/blob/master/channel/pools.go#L486C21-L486C34

https://github.com/Hucaru/Valhalla/blob/master/channel/monster.go#L279-L281